conan/tools/scons/sconsdeps.py

```diff
@@ -0,0 +1,62 @@
+from jinja2 import Template
+from conan.tools._check_build_profile import check_using_build_profile
+from conans.model.new_build_info import NewCppInfo
+from conans.util.files import save
+
+
+class SConsDeps:
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._ordered_deps = []
+        self._generator_file = 'SConscript_conandeps'
+        check_using_build_profile(self._conanfile)
+
+    @property
+    def ordered_deps(self):
+        if not self._ordered_deps:
```

Comment:
This doesn't work when no deps are defined. Better initialize it with ``None`` in the constructor, and compare with ``if self._ordered_deps is None``